plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -259,7 +263,7 @@ public void showOptInDialog(final Shell parentShell) {
       display.syncExec(new Runnable() {
         @Override
         public void run() {
-          new OptInDialog(findShell(parentShell)).open();
+          new OptInDialog(findShell(parentShell), dialogOpenStatus).open();
```

Comment:
> why is it the dialog's responsibility to manage itself?

I thought about that too. If you think it's okay to set `dialogOpenStatus` only once and don't care about resetting it back to false when the dialog closes, I'll go with that. It should work.

I'll also chance the method name. Good point. `verifyAndAskConsent`? I'd want to convey that a dialog can be open.